src/main/telemetry/crsf.c

```diff
@@ -242,6 +245,46 @@ void crsfFrameBatterySensor(sbuf_t *dst)
     sbufWriteU8(dst, batteryRemainingPercentage);
 }
 
+#ifdef USE_BARO
+// pack altitude in decimeters into a 16-bit value.
+// Due to strange OpenTX behavior of count any 0xFFFF value as incorrect, the maximum sending value is limited to 0xFFFE (32766 meters)
+// in order to have both precision and range in 16-bit 
+// value of altitude is packed with different precision depending on highest-bit value.
+// on receiving side:
+// if MSB==0, altitude is sent in decimeters as uint16 with -1000m base. So, range is -1000..2276m.
+// if MSB==1, altitude is sent in meters with 0 base. So, range is 0..32766m (MSB must be zeroed).
+// altitude lower  than -1000m is sent as zero   (should be displayed as "<-1000m" or something).
+// altitude higher than 32767m is sent as 0xfffe (should be displayed as ">32766m" or something).
+// range from 0 to 2276m might be sent with dm- or m-precision. But this function always use dm-precision.
+static inline uint16_t get_altitude_packed(int32_t altitude_dm) 
+{
+    enum { 
+        ALT_MIN_DM = 10000,                     //minimum altitude in dm.
+        ALT_TRESHOLD_DM = 0x8000 - ALT_MIN_DM,  //altitude of precision changing in dm
+        ALT_MAX_DM  = 0x7ffe * 10 - 5,          //maximum altitude in dm
+    };
+    if (altitude_dm < -ALT_MIN_DM) {   //less than minimum
```

Comment:
This is a little confusing. is the minimum 10000 or -10000?

Reply:
Enum with such a calculation and inside function is definitely confusing.

Reply:
I've renamed the constant to be less confusing.
Regarding calculations in enum: it's not a classical enum. It's declaration of local (for the function) constants. What's confusing in it? Of course I could declare just a number after calculating it manually. But I think it's less confusing to show how the constant is calculated. And I don't see any reason why these constants should be global for the module if they are used only in the function.